File: Java/LeetCode/Faulty Keyboard.java
// https://leetcode.com/problems/faulty-keyboard/

class Solution {
    public String finalString(String s) {
        StringBuilder finalStr = new StringBuilder();

        for (int i = 0; i < s.length(); ++i) {
            if (s.charAt(i) == 'i') {
                finalStr.reverse();
            } else {
                finalStr.append(s.charAt(i));
            }
        }

        return finalStr.toString();
    }
}

File: Kotlin/LeetCode/Faulty Keyboard.kt
// https://leetcode.com/problems/faulty-keyboard

class Solution {
    fun finalString(s: String): String {
        val finalStr = StringBuilder();

        for (c in s) {
            if (c == 'i') {
                finalStr.reverse();
            } else {
                finalStr.append(c);
            }
        }

        return finalStr.toString();
    }
}
